lib/appsignal/hooks/active_job.rb

```
@@ -86,7 +103,9 @@ def execute(job)
         private
 
         def transaction_set_error(transaction, exception)
-          return if Appsignal.config[:activejob_report_errors] == "none"
+          # Only report errors when the config option is set to "all".
+          # To report errors on discard, see the `after_discard` callback.
+          return unless Appsignal.config[:activejob_report_errors] == "all"
```

The Oban instrumentation has a similar scenario, where it is only possible to tell if a job's failure caused it to be discarded in certain Oban versions (see the notice in ["Error reporting"](https://docs.appsignal.com/elixir/integrations/oban.html#error-reporting) in the Oban docs)

For unsupported versions, we make `"discard"` in Oban behave like `"all"` -- this seems preferable to me than making it behave like `"none"`, which will silently swallow all errors.

(since there is an explicit version check here, unlike in Oban, a warning could be logged as well when `"discard"` is used with an Active Job version that does not support it)

```suggestion
        if Appsignal::Hooks::ActiveJobHook.version_7_1_or_higher?
          # On ActiveJob versions that support `after_discard`,
          # only report errors when the config option is set to "all".
          # To report errors on discard, see the `after_discard` callback.
          return unless Appsignal.config[:activejob_report_errors] == "all"
        else
          return if Appsignal.config[:activejob_report_errors] == "none"
```

---

As much as I'd like to do that in the integration itself, I can't make active job play nice in testing. So I'm going to move this logic to the config section and overwrite it with "all" if active job is lower than 7.1.

---

Yeah, that works. We had an "overrides" source of some sort in the config for this sort of stuff, right?

---

Yes, exactly what I'm using.